io-engine/src/grpc/v0/mayastor_grpc.rs

```diff
@@ -1546,9 +1551,13 @@ impl mayastor_server::Mayastor for MayastorSvc {
                 let nexus = nexus_lookup(&args.uuid)?;
 
                 match args.action {
-                    0 => nexus.offline_child(&args.uri).await,
+                    0 => {
+                        nexus.fault_child(&args.uri, FaultReason::Offline).await
+                    }
                     1 => nexus.online_child(&args.uri).await,
-                    2 => nexus.retire_child(&args.uri).await,
+                    2 => {
+                        nexus.fault_child(&args.uri, FaultReason::IoError).await
```

Comment:
What's the reasoning to choose IoError as fault reason here? Since the API is not changing, the Retire action in ChildAction lands here, which presumably means this child isn't intended to be used again?

Reply:
It was IoError before, I just moved the code around a bit.
We need to discuss this matter with Tiago.